# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.h
# Commit: cda8bed19391
# Full Hash: cda8bed193914e4e6bedf875aecd9abbcc74606e
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2024-08-02 15:37:12
# Description:
#   Bug 1910880 - Use a real JIT exit frame for Wasm JitEntry stub. r=rhunt
#   
#   JS JIT callers can call Wasm code either directly from Ion or through the JitEntry stub.
#   In the JitEntry case, we didn't have a real JS JIT exit frame. To make JS frame iteration
#   still work in this case, a special `JSJitToWasm` frame type was used to transition from Wasm
# ==============================================================================

diff -r 9b92adc2e91f -r cda8bed19391 js/src/wasm/WasmFrameIter.h
--- a/js/src/wasm/WasmFrameIter.h	Fri Aug 02 06:59:32 2024 +0000
+++ b/js/src/wasm/WasmFrameIter.h	Fri Aug 02 07:14:08 2024 +0000
@@ -69,7 +69,8 @@
   Frame* fp_;
   Instance* instance_;
   uint8_t* unwoundCallerFP_;
-  mozilla::Maybe<jit::FrameType> unwoundJitFrameType_;
+  // Whether unwoundCallerFP_ is a JS JIT exit frame.
+  bool hasUnwoundJitFrame_ = false;
   Unwind unwind_;
   void** unwoundAddressOfReturnAddress_;
   uint8_t* resumePCinCurrentFrame_;
@@ -98,7 +99,6 @@
   void** unwoundAddressOfReturnAddress() const;
   bool debugEnabled() const;
   DebugFrame* debugFrame() const;
-  jit::FrameType unwoundJitFrameType() const;
   bool hasUnwoundJitFrame() const;
   uint8_t* unwoundCallerFP() const { return unwoundCallerFP_; }
   Frame* frame() const { return fp_; }